### ISSUE TITLE
Duplicate / multiple of the same GameResource shouldn't create multiple New.. buttons

### ISSUE THREAD

**Issue by gio3k** [Facepunch/sbox-issues#3020]
### Describe the bug

(This might not seem like a bug - but what else do I label it as?)

I share my GameResource code between my tool & game - this unfortunately makes multiple Create.. buttons

### To Reproduce

1. Have a GameResource in a game...
2. Put that GameResource in a tool too.

### Expected behavior

Only 1 button per file extension preferably.

Simple fix:

(EditorBase/code/CreateAsset.cs, line 74, last foreach statement)
```cs
foreach ( var gameResource in TypeLibrary.GetAttributes<GameResourceAttribute>()
    .OrderBy( x => x.Name )
    .DistinctBy( x => x.Extension ) )
```

### Media/Files

![image](https://user-images.githubusercontent.com/91263142/224919573-40df57a6-fc09-4e4a-afc3-2a5dc20c6b70.png)

**Comment by TheBoxTrot** [comment]
Yeah tricky issue to solve the real problem is being able to refer to gameresources from games inside tools so you don't have to maintain two copies

**Comment by lotuspar** [comment]
Yeah - tool to game data sharing is a huge pain point for me right now

(I’m not sure it could even ever have a perfect solution with the way load contexts work)
